saltproc/saltproc.py

```diff
@@ -265,108 +500,166 @@ def write_mat_file(self):
-
-        # write the processed material to mat file for next run
-        self.write_mat_file()
+        self.core = self.read_dep()
+
+        # make core values mass by multiplying by volume
+        self.core[self.driver_mat_name] = self.core[self.driver_mat_name] * \
+            self.driver_vol
+        try:
+            self.core[self.blanket_mat_name] = self.core[self.blanket_mat_name] * \
+                self.blanket_vol
+        except:
+            self.core[self.blanket_mat_name] = np.zeros(
+                self.number_of_isotopes)
+        # save pre-processing core mass
+        self.core_mass = {}
+        for key, val in self.core.items():
+            self.core_mass[key] = sum(val)
+
+        # record the depleted composition before reprocessing
+        self.driver_before_db[self.current_step,
+                              :] = self.core[self.driver_mat_name]
+        self.blanket_before_db[self.current_step,
+                               :] = self.core[self.blanket_mat_name]
+
+        # waste / fissile tank db initialization
+        self.waste_tank_db[self.current_step,
+                           :] = self.waste_tank_db[self.current_step-1, :]
+        self.fissile_tank_db[self.current_step,
+                             :] = self.fissile_tank_db[self.current_step-1, :]
+
+        # removal first
+        for group, scheme in self.rep_scheme.items():
+            iso_indx = self.find_iso_indx(scheme['element'])
+            if scheme['to'] == 'waste':
+                # things to dump out
+                self.waste_tank_db[self.current_step, :] += self.remove_iso(iso_indx,
+                                                                            scheme['eff'], scheme['from'])
+                print('REMOVING %f kg of %s FROM %s' %
+                      (self.removed_qty, group, scheme['from']))
+            else:
+                continue
+
+        # get mass lacking in reactor after reprocessing out waste
+        self.core_space = {}
+        for mat, val in self.core.items():
+            self.core_space[mat] = self.core_mass[mat] - sum(val)
+
+        # move things around (fissile to driver)
+        for group, scheme in self.rep_scheme.items():
+            if scheme['to'] != 'waste' and scheme['from'] != 'fertile':
+                removed = self.remove_iso(
+                    iso_indx, scheme['eff'], scheme['from'])
+                print('REMOVING %f kg of %s FROM %s' %
+                      (self.removed_qty, group, scheme['from']))
+                # if the movement flow is more than the space in the destination,
+                if sum(removed) > self.core_space[scheme['to']]:
+                    removed_comp = removed / sum(removed)
+                    self.core[scheme['to']] += removed_comp * \
+                        self.core_space[scheme['to']]
+                    print('MOVING %f kg of %s FROM %s TO %s ' % (
```

Comment:
Why capital letters? 
Please talk to me about how to use a reasonable logging library.